### PR TITLE
feat(hybridcloud) Move reply by email to outboxes

### PR DESCRIPTION
**Pull request by markstory**
Because replies come to a singular spot (control silo) we need to delegate to the region silos to actually process the email replies. Using an outbox for this ensures that all messages we receive will be delivered to the region in the order they were received. Email reply outboxes have incrementing object_identifier values to prevent messages to the same group being coalesced together.

I've updated the comment saving logic to prevent duplicate messages from the same user. This could prevent legitimate updates, but I don't see another way to make this operation idempotent.

Refs HC-522

### CODE REVIEW

File: src/sentry/tasks/email.py [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
 logger = logging.getLogger(__name__)
 
 
-def _get_user_from_email(group, email):
-    from sentry.models.user import User
-
-    # TODO(dcramer): we should encode the userid in emails so we can avoid this
-    for user in User.objects.filter(email__iexact=email):
+def _get_user_from_email(group: Group, email: str) -> Optional[RpcUser]:
+    for user in user_service.get_many_by_email(emails=[email]):
```
Comment (scefali) [inline src/sentry/tasks/email.py:16]: @markstory how is this going to handle the `email__iexact` part of the query?

File: src/sentry/web/forms/__init__.py [hunk 2]
```diff
@@ -9,7 +9,18 @@ class NewNoteForm(forms.Form):
...
-    def save(self, group, user, event=None):
+    def save(self, group, user):
+        qs = Activity.objects.filter(
+            group=group,
+            project_id=group.project_id,
+            user_id=user.id,
+            type=ActivityType.NOTE.value,
+            data=self.cleaned_data,
+        )
+        # Prevent duplicate comments, this is necessary for outbox based
```
Comment (scefali) [inline src/sentry/web/forms/__init__.py:20]: @markstory should we push this logic to the manager?

Comment (markstory) [reply src/sentry/web/forms/__init__.py:20]: We could move it there, I didn't want to move too much around at once though.

**Review by iambriccardo — COMMENTED** [review]
LGTM but I lack context for approving

**Review by mrduncan — APPROVED** [review]

**Review by GabeVillalobos — APPROVED** [review]

**Pull request merged**